src/argilla/server/contexts/validations.py

```diff
@@ -0,0 +1,25 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from argilla.server.contexts import accounts
+from argilla.server.models import User, Workspace
+
+
+def can_user_access_to_workspace(user: User, workspace: Workspace):
```

Comment:
I think we need more granularity with the permissions. Maybe something like `is_authorized(user, "create", workspace)` or something similar, "access" is a little bit ambiguous.

In any case we can have a discussion about this.

Reply:
I have done some changes here. We have no functional pattern matching in python so, this kind of function definition could grow easily when we need to authorize several resources for a user. Anyway, let's discuss this from here on Monday.